packages/vulcan-backoffice/lib/components/BackofficeLayout.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect, useRef } from 'react';
+import {
+  getAuthorizedMenuItems,
+  menuItemProps,
+  registerComponent,
+  withCurrentUser,
+  Components,
+} from 'meteor/vulcan:core';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const MenuItem = (
+  { name, label, path, onClick, labelToken, LeftComponent, RightComponent },
+  { intl }
+) => {
+  let Wrapper = React.Fragment;
+  if (path) {
+    const LinkToPath = ({ children }) => <Link to={path}>{children}</Link>;
+    Wrapper = LinkToPath;
+  }
+  return (
+    <Wrapper key={name}>
+      <li
+        //selected={path && router.isActive(path)}
+        onClick={onClick}>
+        {LeftComponent && <LeftComponent />}
+        <span>{label || intl.formatMessage({ id: labelToken })}</span>
+        {RightComponent && <RightComponent />}
+      </li>
+    </Wrapper>
+  );
+};
+
+MenuItem.propTypes = {
+  ...menuItemProps,
+  // parent can pass another onClick callback
+  // eg to close the menu
+  afterClick: PropTypes.func,
+};
+
+const PageLayout = styled.div`
```

Comment:
Instead you should have smth like `const PageLayout = (props) => <div {...props}>{children}</div>; registerComponent({name: "VulcanBackofficePageLayout", component: PageLayout});`. Then only in Bootstrap/MaterialUI you can replace the component and add style to it.

Reply:
This sound a bit unnatural in React but that's how we allow a package oriented architecture in Vulcan with easy customization of components.

Reply:
Also in material UI we don't use Styled components either but Material UI own styling solution (syntax is exactly the same so you can probably copy paste most of you code and just change the import)

Reply:
Ok for the first point, we are going to change the code. But what are the adventages of using material-ui own styling? It makes inconsistencies in the code, no? It's easier to have the same (or almost) code for both bootstrap and material-ui.

Reply:
For React component that may be true but for the CSS because they simply don't use the same approach. Material UI uses CSS-in-JS, while Bootstrap doesn't. We can't force people to setup styled-components or CSS-in-JS in their own app, Vulcan's users usually uses only one of those framework so they don't care about consistency, it just affects us as contributors. You are right that it duplicates our code a bit but for the end users it feels way more intuitive. 

I worry more about the discrepancies we can have within the components logic, that are actually difficult to solve, than about the CSS.

Reply:
Using material UI solution is consistent within the `vulcan:ui-material` package so it's ok